tmp.py

```diff
@@ -0,0 +1,81 @@
+"""Test atlases masking."""
+
+import numpy as np
+from rich import print
+
+from nilearn import datasets
+from nilearn._utils.exceptions import DimensionError
+from nilearn.datasets import (
+    fetch_atlas_aal,
+    fetch_atlas_basc_multiscale_2015,
+    fetch_atlas_destrieux_2009,
+    fetch_atlas_difumo,
+    fetch_atlas_harvard_oxford,
+    fetch_atlas_juelich,
+    fetch_atlas_msdl,
+    fetch_atlas_pauli_2017,
+    fetch_atlas_schaefer_2018,
+    fetch_atlas_talairach,
+    fetch_atlas_yeo_2011,
+)
+from nilearn.maskers import NiftiLabelsMasker
+
+functions = {
+    fetch_atlas_destrieux_2009: None,
+    fetch_atlas_harvard_oxford: ["cort-maxprob-thr0-1mm"],
+    fetch_atlas_juelich: ["maxprob-thr0-1mm"],
+    fetch_atlas_talairach: ["gyrus"],
+    fetch_atlas_aal: None,
+    fetch_atlas_schaefer_2018: None,
+    fetch_atlas_basc_multiscale_2015: [None, None, True, 1, 64, "sym"],
+    fetch_atlas_yeo_2011: None,
+    fetch_atlas_difumo: None,
+    fetch_atlas_msdl: None,
+    fetch_atlas_pauli_2017: None,
+}
+
+
+def main():
+    """Test atlases masking."""
+    dataset = datasets.fetch_development_fmri(n_subjects=1)
+    func_filename = dataset.func[0]
+
+    for f, args in functions.items():
+        print(f)
+
+        atlas = f(*args) if args else f()
+
+        if "labels" in atlas:
+            labels = atlas.labels
+        elif "rsn_indices" in atlas:
+            labels = atlas.rsn_indices
+        elif f == fetch_atlas_basc_multiscale_2015:
+            labels = range(64)
+        elif f == fetch_atlas_yeo_2011:
+            labels = range(17)
+
+        if f == fetch_atlas_schaefer_2018:
+            labels = np.insert(labels, 0, "Background")
```

Comment:
I would also add this for schaefer's atlas since its label are bytestring for some reason!
```
labels = [label.decode("utf-8") for label in labels]
```
And I think you might need to add "Background" label for AAL atlas too since it doesn't have it originally.

Reply:
I totally agree on both.

But I think that those kind of additions are independent of the bug fix and should go in a different PR and as concerted effort to "standardise" the fetchers output, but as this may change what some fetchers return I suspect we will have to go through some deprecation cycles.

Reply:
I actually added a `_sanitize_labels` methods in to at least handle the case where labels are an iterable of bytes.